lib/subscription-connection.js

```diff
@@ -30,13 +31,17 @@ module.exports = class SubscriptionConnection {
     this.subscriber = subscriber
     this.subscriptionContexts = new Map()
 
-    this.socket.on('message', (message) => {
-      this.handleMessage(message).catch(e => {
-        this.fastify.log.error(e)
+    this.socket.on('error', this.handleConnectionClose.bind(this))
+    this.handleConnection()
+  }
+
+  async handleConnection () {
+    for await (const message of on(this.socket, 'message')) {
+      this.handleMessage(message).catch(err => {
```

Comment:
you need to await this

Reply:
> The test hangs after the subscribe https://github.com/mcollina/fastify-gql/pull/228/files#diff-d0a65557fba21a77676fdc6c8ab721c7L105
> 
> ```
> subscription {
>       notificationAdded {
>       id
>       message
>    }
>  }
> ```

As said earlier, I don't understand why but if I await this line it blocks everything

Reply:
I think it's because the `handleMessage` function doesn't return if a subscription is called with `handleGQLStart`: https://github.com/mcollina/fastify-gql/blob/master/lib/subscription-connection.js#L105

Reply:
Do you have any suggestion ? This solution doesn't work because of the `handleMessage` will not return until the subscription is dropped, therefore blocking the iteration if we await `handleMessage`.

Reply:
The subscription context is store in `this.subscriptionContexts` and deleted if the connection is closed so I've removed the `await` before `handleGQLStart`, is this safe ?

Reply:
This is ok, yes.